src/Utilities/Math.hpp

```diff
@@ -79,3 +79,24 @@ template <typename T, Requires<std::is_arithmetic<T>::value> = nullptr>
 auto invcbrt(const T& arg) noexcept {
   return static_cast<T>(1.0) / cbrt(arg);
 }
+
+namespace sgn_detail {
+template <typename T>
+constexpr int sgn(const T& val, std::true_type /*is_signed*/) noexcept {
+  return static_cast<int>(static_cast<bool>(static_cast<T>(0) < val)) -
+         static_cast<int>(static_cast<bool>(val < static_cast<T>(0)));
```

Comment:
Is the`bool` cast necessary? I thought the comparison operator for any of the types this works for would return bools.

Reply:
I'll check, good point.